### PR TITLE
Switch to generic third party onboarding

### PR DESCRIPTION
**Pull request by Davidde94**

### CODE REVIEW

File: Sources/SpartaLib/Users/Hunter/Hunter.swift [hunk 1]
```diff
@@ -10,9 +10,21 @@
...
-    case Email = 1
-    case Facebook = 2
+enum HunterLoginType: String, Codable {
+    
+    case Email = "email"
+    case Facebook = "facebook"
+    
+    init?(rawValue: String) {
+        switch rawValue.lowercased() {
+        case "email":
```
[the remark this answers is not in the document]
Comment (Davidde94) [reply Sources/SpartaLib/Users/Hunter/Hunter.swift]: This is an extra layer of protection. Notice the `lowercased()` function call (e.g. now both "Facebook" and "FACEBOOK" would be equivalent to "facebook" and work as such. After some consideration, however, the benefits aren't worth the performance hit.

File: Tests/SpartaLibTests/HunterTests.swift [hunk 2]
```diff
@@ -15,7 +15,7 @@ class HunterTests: XCTestCase {
     func testCreationFromInfoStruct() {
         
         do {
-            let facebookOnboardingInfo = HunterFacebookOnboardingInfo(forename: "David", surname: "Evans", email: nil, auth: "123456789", secret: "blah")
+            let facebookOnboardingInfo = HunterThirdPartyOnboardingInfo(forename: "David", surname: "Evans", email: nil, auth: "david_evans94@icloud.com", secret: "blah", loginType: .Facebook)
```
[the remark this answers is not in the document]
Comment (Davidde94) [reply Tests/SpartaLibTests/HunterTests.swift]: No idea what you're on about - it is done separately.

[the remark this answers is not in the document]
Comment (RyanMKrol) [reply Tests/SpartaLibTests/HunterTests.swift]: Testing that a dictionary gets decoded into a codable object is one complete test. Testing that, and then testing that passing it through into a user object is an integration test.

[the remark this answers is not in the document]
Comment (Davidde94) [reply Tests/SpartaLibTests/HunterTests.swift]: Look at the test haha, they're 2 separate tests in the code, therefore unit tests...

**Pull request merged**